[alea] compilation failure with gcc-7.1

alea does not compile with gcc-7.1 on cori due to the following error

```
/global/homes/j/jkleinh/research/libraries/ALPSCore/alea/include/alps/alea/complex_op.hpp:207:28: error: invalid static_cast from type '<unresolved overloaded function type>' to type 'bool (*)(double)'
                            static_cast<bool(*)(T)>(std::isnan));
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/alea/include/alps/alea/complex_op.hpp
+++ b/alea/include/alps/alea/complex_op.hpp
@@ -203,14 +203,12 @@ public:
 
     friend bool isnan(complex_op x)
     {
-        return std::any_of(&x.vals_[0][0], &x.vals_[2][0],
-                           static_cast<bool(*)(T)>(std::isnan));
+        return std::any_of(&x.vals_[0][0], &x.vals_[2][0], [](T y) {return std::isnan(y); });
     }
 
     friend bool isfinite(complex_op x)
     {
-        return std::all_of(&x.vals_[0][0], &x.vals_[2][0],
-                           static_cast<bool(*)(T)>(std::isfinite));
+        return std::all_of(&x.vals_[0][0], &x.vals_[2][0], [](T y) {return std::isfinite(y); });
     }
 
     friend bool isinf(complex_op x)
@@ -220,8 +218,7 @@ public:
         if (isnan(x))
             return false;
 
-        return std::any_of(&x.vals_[0][0], &x.vals_[2][0],
-                           static_cast<bool(*)(T)>(std::isinf));
+        return std::any_of(&x.vals_[0][0], &x.vals_[2][0], [](T y) {return std::isinf(y); });
     }
 
     friend complex_op abs(complex_op x) { return sqrt(abs2(x)); }

